src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1372,12 +1383,14 @@ bool ColorControlServer::moveSaturationCommand(app::CommandHandler * commandObj,
 bool ColorControlServer::moveToSaturationCommand(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
                                                  const Commands::MoveToSaturation::DecodableType & commandData)
 {
-    uint8_t saturation      = commandData.saturation;
-    uint16_t transitionTime = commandData.transitionTime;
-    uint8_t optionsMask     = commandData.optionsMask;
-    uint8_t optionsOverride = commandData.optionsOverride;
-    EndpointId endpoint     = commandPath.mEndpointId;
-    Status status           = Status::Success;
+    uint8_t saturation        = commandData.saturation;
+    uint16_t transitionTime   = commandData.transitionTime;
+    uint16_t numberOfSteps    = transitionTime == 0 ? 1 : transitionTime;
```

Comment:
For readability you should define/unit those just before you use them, at least after the block of commandData has been assigned.

Same comment applies in other locations

Reply:
I've removed these variables and just performed the calculation where they get used.